### PR TITLE
fix: duplicate flow in API

### PR DESCRIPTION
**Pull request by hiteshbedre**

### CODE REVIEW

File: src/main/scala/ai/privado/exporter/DataflowExporter.scala [hunk 1]
```diff
@@ -104,19 +99,23 @@ class DataflowExporter(cpg: Cpg, dataflowsMap: Map[String, Path], taggerCache: T
...
       var sinkId = sourceModel.sinkId
       val sinkAPITag = dataflowsMap(sourceModel.pathId).elements.last.tag
         .filter(node => node.name.equals(Constants.apiUrl + sourceModel.sinkId))
-      if (sinkAPITag.nonEmpty) {
-        sinkId += "#_#" + sinkAPITag.value.l.mkString(",")
-      }
+      if (!sinkApiUrls.contains(sinkId))
+        sinkApiUrls(sinkId) = mutable.Set()
+      for (url <- sinkAPITag)
+        sinkApiUrls(sinkId).add(url.value)
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/exporter/DataflowExporter.scala:111]: There should be `addAll` kind of function, use that

File: src/main/scala/ai/privado/exporter/DataflowExporter.scala [hunk 2]
```diff
@@ -70,23 +70,18 @@ class DataflowExporter(cpg: Cpg, dataflowsMap: Map[String, Path], taggerCache: T
...
-      val sinkIdAfterSplit = sinkId.split("#_#")
+    def convertSink(sourceId: String, sinkId: String, sinkPathIds: List[String], urls: mutable.Set[String]) = {
 
       // Special case for API type of nodes
-      val apiUrl = ruleCache.getRuleInfo(sinkIdAfterSplit(0)) match {
-        case Some(rule) if rule.nodeType.equals(NodeType.API) & sinkIdAfterSplit.size >= 2 =>
-          sinkIdAfterSplit(1).split(",").toList
-        case _ => List[String]()
-      }
+      val apiUrl = if (urls.size > 0) urls.toList else List[String]()
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/exporter/DataflowExporter.scala:76]: `urls.empty` or `isEmpty` can be used

File: src/main/scala/ai/privado/exporter/DataflowExporter.scala (hunk 1 above)
Comment (hiteshbedre) [reply src/main/scala/ai/privado/exporter/DataflowExporter.scala]: Addressed here: https://github.com/Privado-Inc/privado-core/commit/4110826932d0e7852b80590ee5c5b98e91c22fdb

File: src/main/scala/ai/privado/exporter/DataflowExporter.scala [hunk 3]
```diff
@@ -70,23 +70,18 @@ class DataflowExporter(cpg: Cpg, dataflowsMap: Map[String, Path], taggerCache: T
     sinkSubCategory: String,
     ruleCache: RuleCache
   ): List[DataFlowSubCategorySinkModel] = {
-    def convertSink(sourceId: String, sinkId: String, sinkPathIds: List[String]) = {
-      val sinkIdAfterSplit = sinkId.split("#_#")
+    def convertSink(sourceId: String, sinkId: String, sinkPathIds: List[String], urls: mutable.Set[String]) = {
```
[the remark this answers is not in the document]
Comment (hiteshbedre) [reply src/main/scala/ai/privado/exporter/DataflowExporter.scala]: Addressed here: https://github.com/Privado-Inc/privado-core/commit/4110826932d0e7852b80590ee5c5b98e91c22fdb

File: src/main/scala/ai/privado/exporter/DataflowExporter.scala (hunk 2 above)
Comment (hiteshbedre) [reply src/main/scala/ai/privado/exporter/DataflowExporter.scala]: Addressed here: https://github.com/Privado-Inc/privado-core/commit/4110826932d0e7852b80590ee5c5b98e91c22fdb

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Pull request merged**